### DISCUSSION TITLE
Will typst be published in the different linux repositories?

### DISCUSSION THREAD

**Discussion by N4tus** [typst/typst#98] [Q&A] [answered]
.

**Comment by suy** [comment]
Distributions package open source software. The authors of that software are (most often) not the ones that publish it into linux distributions. You are asking the wrong people.

**Comment by pbsds** [comment, marked as answer]
It was on hackernews, its bound to get packaged. The efforts i found with a quick search:

* [nixpkgs](https://github.com/NixOS/nixpkgs/pull/222437)
* [AUR](https://aur.archlinux.org/packages/typst)
* [Scoop](https://github.com/ScoopInstaller/Main/blob/master/bucket/typst.json)

Edit: [Typst on repology](https://repology.org/project/typst/versions)

**Reply by ondohotola** [reply]
And `homebrew`?

**Reply by pbsds** [reply]
Newer used it, but [this](https://github.com/Homebrew/homebrew-core) seems like the main repo/"tap". 
I find no typst pull request, so try making an issue there.

Seems like a [MacPorts](https://ports.macports.org/port/typst/details/) package has been created sine i last posted.

**Reply by user202729** [reply]
The AUR package got moved to the community repository: [Arch Linux - typst 23.03.21.2-1 (x86_64)](https://archlinux.org/packages/community/x86_64/typst/)

**Reply by classabbyamp** [reply]
Void has a package now - void-linux/void-packages#43185

**Reply by PgBiel** [reply]
OpenSUSE also has a package (albeit not fully up-to-date right now): https://build.opensuse.org/package/show/openSUSE:Factory/typst

**Reply by drupol** [reply]
Since Typst is available on Nix, it is available on every single Linux out there.

**Reply by l3d00m** [reply]
The arch package now has been moved to the extra repository: https://archlinux.org/packages/extra/x86_64/typst/

**Comment by ondohotola** [comment]
Peter,

I am sure this will pitch up on Homebrew.

I have no idea how to make pull requests or "issues" on Homebrew.

el


On 23/03/2023 23:41, Peder Bergebakken Sundt wrote:
> Newer used it, but this <https://github.com/Homebrew/homebrew-core>
> seems like the main repo/"tap".
> I find no typst pull request, so try making an issue there.
> 
> Seems like a MacPorts <https://ports.macports.org/port/typst/details/>
> package has been created sine i last posted.
[...]


-- 
Dr. Eberhard W. Lisse   \         /       Obstetrician & Gynaecologist
***@***.***             / *      |  Telephone: +264 81 124 6733 (cell)
PO Box 8421 Bachbrecht  \      /  If this email is signed with GPG/PGP
10007, Namibia           ;____/ Sect 20 of Act No. 4 of 2019 may apply

**Reply by Altdarkzodiac** [reply]
Any luck yet i can help

**Comment by KronosTheLate** [comment]
What about `apt`

**Reply by drupol** [reply]
Since it's available on `nixpkgs`, you can have it working on every single linux using Nix.

**Reply by KronosTheLate** [reply]
Cool! I have never heard about nix. Does it replace apt?

**Reply by drupol** [reply]
Nix is a universal package manager (not based on containers). It should not replace `apt` on your distribution because your distribution is built "around" `apt`.  That said, there is a distribution built around Nix, called NixOS.

I invite you to install Nix using this procedure: https://github.com/DeterminateSystems/nix-installer and play with it

Feel free to ask for help

**Reply by KronosTheLate** [reply]
Sounds like a nice tool. It is not a priority to mess around with it right now, but thanks a lit for the helpfullness. Perhaps I will revisit this thread some time ^_^

**Reply by extua** [reply]
I have proposed Typst for packaging in Debian [here](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1060888), but it needs someone to volunteer to maintain it.

**Reply by tgross35** [reply]
If anyone is able to help, I wrote up the steps needed to make this happen here: https://github.com/typst/typst/issues/3679

**Comment by ondohotola** [comment]
It is YAPM (yet another package manager), while a bit more current on
older linux systems.

I really rather preferred if APT and RPM packages were generated,
perhaps on external repositories (for APT at least, don't know if that
works form RPM), so an APT UPGRADE will catch that.

Gruss aus dem Süden

**Comment by rusty-snake** [comment]
If anyone cares, I created a spec file for typst in my fedora-extra repo. https://github.com/rusty-snake/fedora-extras/blob/main/typst/typst.spec

**Comment by freundTech** [comment]
Typst is also packaged on [conda-forge](https://anaconda.org/conda-forge/typst), so it can be installed on any Linux distro, Mac OS and Windows using `conda` or your favorite conda-compatible package manager (such as micromamba or pixi)
